web_infrastructure/gfdeploy

```diff
@@ -0,0 +1,193 @@
+        f = f[:last]
+    return f
+
+def deploy(url, contextroot, name, filepath, force, user, password):
+    params = {'contextroot' : contextroot, 'name' : name, 'force':str(force)}
+    requests.packages.urllib3.disable_warnings()
+    f = MultipartEncoder(
+        fields={'contextroot':contextroot, 'name':name, 'force':str(force), 'id':(getFilename(filepath), open(filepath, 'rb'), 'application/octet-stream')}
+    )
+    r = requests.post(url, verify=False, data=f, headers={'Content-Type':f.content_type, 'X-Requested-By':'ansible'}, auth=(user,password))    
+    if r.status_code == 200:
+        return True, r.status_code
+    return False, r.status_code
+
+def undeploy(url, user, password):
+    requests.packages.urllib3.disable_warnings()
+    r = requests.delete(url, headers={'X-Requested-By':'ansible'}, verify=False, auth=(user,password))
+    return (True, r.status_code) if r.status_code == 200 else (False, r.status_code)
+
+def protocol(secure):
+    return 'https://' if bool(secure) == True else 'http://'
+
+def getUndeployUrl(host, port, secure, appName):
+    url = protocol(secure)
+    url += host + ':' + port + '/management/domain/applications/application/' + appName
+    return url
+    
+
+def getDeployUrl(host, port, secure):
+    url = protocol(secure)
+    url += host + ':' + port + '/management/domain/applications/application'
+    return url
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+	    contextroot=  	dict(required=False),
+	    name=  			dict(required=True),
+	    filepath=		dict(required=False),
+        secure=			dict(required=False, default=True, choice=BOOLEANS),
+	    host=			dict(required=False, default='localhost'),
+	    port=			dict(required=False, default='4848'),
+	    user=			dict(required=True),
+	    password=		dict(required=True),	    
+	    force=			dict(required=False, default=False),
+	    state =			dict(required=False, default='present', choices=['present','absent'])
+        )
+    )
+
+    contextroot = module.params['contextroot']
+    name 	= module.params['name']
+    filepath 	= module.params['filepath']
+    force 	= module.params['force']
+    state 	= module.params['state'] 
+    host	= module.params['host']
+    port	= module.params['port']
+    user	= module.params['user']
+    password	= module.params['password']
+    secure 	= module.params['secure']
+
+    if state == 'absent':
+        url = getUndeployUrl(host,port,secure, name)
+        undeployed = undeploy(url, user, password)
+        module.exit_json(location=url, changed=undeployed[0], response_code=undeployed[1])
+    url = getDeployUrl(host,port,secure)
+    deployed = deploy(url, contextroot, name, filepath, force, user, password)	
+    module.exit_json(location=url, changed=deployed[0], response_code=deployed[1]) 
+
+
+from ansible.module_utils.basic import *
+import re
+import requests
+from requests.auth import HTTPBasicAuth
+from requests_toolbelt import MultipartEncoder
+import ntpath
```

Comment:
Only the `from ansible.module_utils.basic import *` import should be down here.  The others should be moved to below the `RETURN` section.